### PR TITLE
ISSUE-58: enable grpc reflection so that grpc_cli can interact with DemoApp

### PR DESCRIPTION
**Pull request by jackyjia**

### CODE REVIEW

File: src/app_demo/should_be_generated/app/RequestReceiver.h [hunk 1]
```diff
@@ -17,6 +17,7 @@ limitations under the License.
 #define SRC_APP_DEMO_SHOULD_BE_GENERATED_APP_REQUESTRECEIVER_H_
 
 #include <INIReader.h>
+#include <grpcpp/ext/proto_server_reflection_plugin.h>
```
Comment (huyumars) [inline src/app_demo/should_be_generated/app/RequestReceiver.h:20]: why not move to cpp to reduce compile time.

File: scripts/installDependencies-macos.sh [hunk 2]
```diff
@@ -29,7 +29,7 @@ if [ -z "$PROTOBUF" ]; then
     mkdir -p build && cd build &&
     # use cmake instead of autogen.sh so that protobuf-config.cmake can be installed
     cmake -Dprotobuf_BUILD_TESTS=OFF -DCMAKE_BUILD_TYPE=Debug .. &&
-    make && make install && make clean && ldconfig
+    make && make install && make clean
```
Comment (huyumars) [inline scripts/installDependencies-macos.sh:32]: you can leave this change to https://github.com/eBay/Gringofts/pull/60 or you have to resolve conficts

File: src/app_demo/should_be_generated/app/RequestReceiver.h (hunk 1 above)
Comment (jackyjia) [reply src/app_demo/should_be_generated/app/RequestReceiver.h]: done

File: scripts/installDependencies-macos.sh (hunk 2 above)
Comment (jackyjia) [reply scripts/installDependencies-macos.sh]: i'll leave all macos-related changes to you. please make sure it compiles.

File: config/app_demo/app_raft_1.ini [hunk 3]
```diff
@@ -15,15 +12,17 @@ wal.dir = ./node_1/rocksdb
 [netadmin]
 ip.port = 0.0.0.0:50065
 
-[store]
+[cluster]
 persistence.type = raft
+cluster.conf = 1#1@0.0.0.0:5253|50055|55055|65055|75055|85055,2@0.0.0.0:5254|50056|55056|65055|75055|85055,3@0.0.0.0:5255|50057|55057|65055|75055|85055
```
Comment (jingyichen1223) [inline config/app_demo/app_raft_1.ini:17]: If the three nodes are running on the same host, is there port conflict on 65055|75055|85055?

File: README.md [hunk 4]
```diff
@@ -104,11 +104,23 @@ This approach requires minimum dependencies on the target OS as all of the depen
...
+   ```
+   Sample output:
+   ```bash
+   connecting to 0.0.0.0:50055
+   code: 200
+   message: "Success"
+
+   Rpc succeeded with OK status
+   ```
+3. Backed by a three-nodes-cluster setup
```
[the remark this answers is not in the document]
Comment (jackyjia) [reply README.md:119]: yes and no. grpc_cli does not support leader discovery.

File: config/app_demo/app_raft_1.ini (hunk 3 above)
Comment (jackyjia) [reply config/app_demo/app_raft_1.ini]: good spot. updated.

**Review by lingjun17 — APPROVED** [review]
LGTM

**Review by lingjun17 — APPROVED** [review]
LGTM

**Pull request merged**